### PR TITLE
Sar 4648

### PR DESCRIPTION
**Pull request by MarjanMubarok**

### CODE REVIEW

File: app/uk/gov/hmrc/vatsignup/controllers/NewVatEligibillityController.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    implicit request =>
+      authorised() {
+        vatNumberEligibilityService.getMtdStatus(vatNumber) map {
+          case AlreadySubscribed => Ok(Json.obj("mtdStatus" -> "AlreadySubscribed"))
+          case Eligible(isMigrated, isOverseas) =>
+            Ok(Json.obj("mtdStatus" -> "MigrationInProgress",
+              "eligibilityDetails" -> Json.obj("isMigrated" -> isMigrated, "isOverseas" -> isOverseas)))
+          case Ineligible => Ok(Json.obj("mtdStatus" -> "Ineligible"))
+          case Inhibited(migratableDates) => Ok(Json.obj("mtdStatus" -> "Inhibited",  "migratableDates" -> Json.toJson(migratableDates)))
+          case MigrationInProgress => Ok(Json.obj("mtdStatus" -> "MigrationInProgress"))
```
Comment (kabuumu) [inline app/uk/gov/hmrc/vatsignup/controllers/NewVatEligibillityController.scala]: Could we take "mtdStatus" out as a constant please as it's reused.

File: app/uk/gov/hmrc/vatsignup/services/VatNumberEligibilityService.scala [hunk 2]
```diff
@@ -32,6 +32,7 @@ import scala.concurrent.{ExecutionContext, Future}
 class VatNumberEligibilityService @Inject()(mandationStatusConnector: MandationStatusConnector,
                                             controlListEligibilityService: ControlListEligibilityService
                                            )(implicit ec: ExecutionContext) {
+
```
Comment (kabuumu) [inline app/uk/gov/hmrc/vatsignup/services/VatNumberEligibilityService.scala]: Is this needed?

File: test/uk/gov/hmrc/vatsignup/controllers/NewVatEligibillityControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+  implicit val system: ActorSystem = ActorSystem()
+  implicit val materializer: ActorMaterializer = ActorMaterializer()
+
+  val testAlreadySubscribedResponse = Json.obj("mtdStatus" -> "AlreadySubscribed")
+  val testEligibleResponse = Json.obj("mtdStatus" -> "MigrationInProgress",
+    "eligibilityDetails" -> Json.obj("isMigrated" -> true, "isOverseas" -> false))
+  val testIneligibleResponse = Json.obj("mtdStatus" -> "Ineligible")
+  val testInhibitedResponse = Json.obj("mtdStatus" -> "Inhibited",  "migratableDates" -> Json.toJson(testMigratableDates))
+  val testMigrationInProgressResponse = Json.obj("mtdStatus" -> "MigrationInProgress")
```
Comment (kabuumu) [inline test/uk/gov/hmrc/vatsignup/controllers/NewVatEligibillityControllerSpec.scala]: If you turn mtdStatus and the values into constants within the class itself, you wouldn't need these constants.

File: test/uk/gov/hmrc/vatsignup/controllers/StoreVatNumberControllerSpec.scala [hunk 4]
```diff
@@ -57,6 +57,7 @@ class StoreVatNumberControllerSpec extends UnitSpec with MockAuthConnector with
 
         val res: Result = await(TestStoreVatNumberController.storeVatNumber()(request))
 
+
```
Comment (kabuumu) [inline test/uk/gov/hmrc/vatsignup/controllers/StoreVatNumberControllerSpec.scala]: Is this needed?

File: conf/app.routes [hunk 5]
```diff
@@ -10,3 +10,5 @@ POST        /sign-up-request/request-id/:requestId/identity-verification
 POST        /sign-up-request/request-id/:requestId/nino                           @uk.gov.hmrc.vatsignup.controllers.StoreNinoWithRequestIdController.storeNino(requestId: String)
 POST        /sign-up-request/request-id/:requestId/transaction-email              @uk.gov.hmrc.vatsignup.controllers.StoreTransactionEmailWithRequestIdController.storeTransactionEmail(requestId: String)
 POST        /sign-up-request/request-id/:requestId/partnership-information        @uk.gov.hmrc.vatsignup.controllers.StorePartnershipInformationWithRequestIdController.storePartnershipInformation(requestId: String)
+
+GET         /sign-up-request/request-id/:requestId/mtdfb-eligibility              @uk.gov.hmrc.vatsignup.controllers.NewVatEligibillityController.checkVatNumberEligibillity(requestId: String)
```
Comment (VLukovski) [inline conf/app.routes]: you forgot to remove this after making the new route in deprecated.routes

File: it/uk/gov/hmrc/vatsignup/controllers/NewVatEligibillityControllerISpec.scala [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        stubOverseasControlListInformation(testVatNumber)
+
+        val res = await(get("/subscription-request/vat-number/:vatNumber/new-mtdfb-eligibility"))
+
+        res should have(
+          httpStatus(INTERNAL_SERVER_ERROR)
+        )
+      }
+    }
+  }
```
Comment (kabuumu) [inline it/uk/gov/hmrc/vatsignup/controllers/NewVatEligibillityControllerISpec.scala]: Could we have one component integration test for each "happy path" flow please. This API needs extensive regression testing as it's going to be used by every journey.

**Pull request merged**